Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -28,10 +30,33 @@ namespace Microsoft.DotNet.Interactive.CSharp
 {
     public class CSharpKernel :
         KernelBase,
-        IExtensibleKernel
+        IExtensibleKernel,
+        ISupportNuget
     {
         internal const string DefaultKernelName = "csharp";
 
+        private DependencyProvider _dependencies;
+
+        private Lazy<IDependencyManagerProvider> _idm;
+
+        // C# delegates on class field initializers can't invoke instance methods or data, so make it a property backed by an unitialized field.
+        // A race can happen, in both the Lazy () and in the field initializer, but it doesn't really matter, since they are both effectively side effect free.
+        private IDependencyManagerProvider GetIdm(ResolvingErrorReport reportError)
+        {
+            if (_dependencies == null)
+            {
+                // This indicates a bug in notbook code. For a Kernel to support nuget
+                // it must have initialized using InitializeDependencyProvider before calling ISupportNuget.Resolve
+                // this should be done in KernelSupportsNugetExtensions.UseNugetDirective<T>(this T kernel)
+                throw new InvalidOperationException("Internal error --- must invoke ISupportNuget.InitializeDependencyProvider before ISupportNuget.Resolve()");
+            }
+            if (_idm == null)
+            {
+                _idm = new Lazy<IDependencyManagerProvider>(() => _dependencies?.TryFindDependencyManagerByKey(Enumerable.Empty<string>(), "", reportError, "nuget"));
```

Comment:
Allowing for the possibility of a race condition and call order dependency here defeats the purpose of using `Lazy`. We should be able to shift the `_idm` field initializer to the constructor.